components/keys/src/lib.rs

```diff
@@ -229,6 +230,28 @@ pub fn data_end_key(region_end_key: &[u8]) -> Vec<u8> {
     }
 }
 
+#[derive(Debug, Display)]
+pub enum Error {
```

Comment:
How about deriving `Fail`?

Reply:
Derive Fail + 1

Reply:
I have made this change on request. I don't see why it's desirable though.